java/vector/src/main/codegen/templates/UnionVector.java

```diff
@@ -688,7 +689,7 @@ public void setType(int index, MinorType type) {
     }
 
     private int getTypeBufferValueCapacity() {
-      return typeBuffer.capacity() / TYPE_WIDTH;
+      return checkedCastToInt(typeBuffer.capacity() / TYPE_WIDTH);
```

Comment:
Maybe apply checkedCastToInt to typeBuffer.capacity() first?

Reply:
In this case, it doesn't really matter since TYPE_WIDTH is one, but the for consistency I've made this capAtMaxInt (I added the method in a previous commit).